config/initializers/rack_attack.rb

```diff
@@ -6,6 +6,10 @@ class Attack
       IPAddr.new(x)
     end
 
+    EMAIL_REGISTRATION_PATHS = ['/sign_up/enter_email', '/es/sign_up/enter_email',
```

Comment:
Random comment for a future potential change - Would be NICE if we could define these by controller so if we add new languages or adjust the URL structure the limits would just follow.

Reply:
Do we have access to URL helpers, or at least `I18n.available_locales` at this point in initialization?

```suggestion
    EMAIL_REGISTRATION_PATHS = [*I18n.available_locales, ''].map { |locale| sign_up_email_path(locale: locale) }
```

Reply:
Also, isn't `/en/sign_up/enter_email` technically a valid path? Should it be included here?

Reply:
Good catch on `/en`, yeah, it should be there as well.

Initializers unfortunately happen before the application is loaded so routes/actions and i18n locales aren't available at this point.  rack-attack has a short section on [matching actions in Rails](https://github.com/rack/rack-attack/blob/main/docs/advanced_configuration.md#match-actions-in-rails). It has a caveat at the top that doesn't inspire confidence, and I don't love the solution since it relies on rescue and doesn't take the HTTP method into account (though it looks like `recognize_path` supports it).

I messed around with `recognize_path` and more dynamic post-initialize solutions and didn't really like any of them. I also don't like the static path-based solution we have currently.